resources/js/Events/Components/Core/EventTree.jsx

```diff
@@ -0,0 +1,281 @@
+class EventTree {
+  constructor(props) {
+    Object.assign(this, props);
+  }
+
+  static make(events, source) {
+    const makeFlatTreeInner = (eventData, parentPath, level = 0) => {
```

Comment:
Please add comment on what the structure looks like.

```suggestion
    /**
     * Given a set of events like:
     *   - top level
     *      - groups: second level
     *          - groups: third level
     *              - data: [event 1, event 2, event 3]
     *  This will return a structure like:
     *  [{top level}, {second level}, {third level}, {event 1}, {event 2}, {event 3}].
     *  Each object in the list will have references to their children and an ID for their parent.
     */
    const makeFlatTreeInner = (eventData, parentPath, level = 0) => {
```